src/statemachine.rs

```diff
@@ -164,121 +139,107 @@ pub struct LinuxData<'a> {
 
 impl <'a> StateData for LinuxData<'a> {
 
-    fn start(&mut self) -> Option<TestState> {
+    fn start(&mut self) -> Result<TestState, RunError> {
         match waitpid(self.current, Some(WaitPidFlag::WNOHANG)) {
-            Ok(WaitStatus::StillAlive) => None,
+            //TODO: Which TestState is correct here?
+            //Ok(WaitStatus::StillAlive) => Ok(None),
```

Comment:
So this function traditionally returned the new `TestState` if it had changed. So if it hadn't changed I returned `None`. So for `Ok(WaitStatus::StillAlive)` you should return the current test state unaltered. If you want it to return the current TestState not a delta. For `Ok(_)` I agree that should be an error as the current behaviour just had it timeout and then kill the test due to the timeout.

Reply:
Returning the current state again means you will have to pass it into `start` as an argument, or you could mimic what you did for `wait` and return a `Result<Option<TestState>, RunError>`